### PR TITLE
Tidy up some Android tests

### PR DESCRIPTION
**Pull request by tamird**
There are still quite a few ignored Android tests kicking around, most of which were added in 445faca8441aae34c91318b6ad9e2049885af8dc, which has a pretty unfortunate commit message.

r? @alexcrichton

### CODE REVIEW

**Comment by tamird** [comment]
A try here would be great

**Comment by tamird** [comment]
try doesn't test on android :(

**Comment by alexcrichton** [comment]
Eh we've got a pretty small queue right now, let's just send this to bors and see what happens!

@bors: r+ 3580084

**Comment by tamird** [comment]
@alexcrichton had some line number drift :(

Give this another go?

File: src/test/run-pass/linkage-visibility.rs [hunk 1]
```diff
@@ -9,7 +9,6 @@
 // except according to those terms.
 
 // aux-build:linkage-visibility.rs
-// ignore-android: FIXME(#10379)
```
Comment (tamird) [inline src/test/run-pass/linkage-visibility.rs]: well this one is still broken

**Comment by tamird** [comment]
@alexcrichton once more?

**Comment by tamird** [comment]
@alexcrichton can you hit the retry? this failed on `net::tcp::tests::fast_rebind` which looks spurious

**Comment by tamird** [comment]
@alexcrichton reverted the `ignore-pretty` change that failed this. once more?

**Comment by tamird** [comment]
Sigh `fast_rebind` again

File: src/libstd/dynamic_lib.rs [hunk 2]
```diff
@@ -133,7 +133,7 @@ mod test {
     use path::Path;
 
     #[test]
-    #[cfg_attr(any(windows, target_os = "android"), ignore)] // FIXME #8818, #10379
```
Comment (tamird) [inline src/libstd/dynamic_lib.rs]: well this one is still broken. removing this commit

**Comment by tamird** [comment]
@alexcrichton almost there!

**Comment by tamird** [comment]
`shootout-k-nucleotide-pipes` still failed :(

@alexcrichton another go?

**Comment by alexcrichton** [comment]
@tamird at this point I think it may be best to start setting up an Android installation locally and testing this out there, bouncing on bors may be starting to take up some time :(

**Comment by alexcrichton** [comment]
So I've run some more of these tests locally and it looks like basically everything that was unignored unfortunately wasn't fixed in the meantime. Perhaps the stylistics tweaks could just be merged?

**Comment by tamird** [comment]
D'oh. #10381 should be reopened, or should a new issue be opened for those? What was the failure?

I'm surprised #9116 failed as well.

By the way, how does one run these locally? I'd be happy to dig deeper in a subsequent PR (I'll rebase this one to keep the stylistics shortly).

**Comment by tamird** [comment]
@alexcrichton updated.

**Comment by alexcrichton** [comment]
@bors: r+ bd5afb406df05997d121c815a3e9f2979236ef2d

To run these locally you'll need to install the NDK and then configure rustc accordingly (I normally just copy the `./configure` line of the bots). You'll then run an emulator in the background which `compiletest` connects to and sends binaries to. [Here's the old wiki page](https://github.com/rust-lang/rust-wiki-backup/blob/master/Doc-building-for-android.md)

**Pull request merged**